T2 cpu credit balance alarm

dagvl opened the pull request:
This implements CPU credit balance alarms when the database is a burstable type. The if/else-like behavior looks a bit hairy. It is taken from https://blog.gruntwork.io/terraform-tips-tricks-loops-if-statements-and-gotchas-f739bbae55f9. The default value is the number of cpu credits (approximately) that a db.t2.micro has when it first starts up.

This is the last pull request I have to make (for now at least).

hectcastro commented on main.tf:
```diff
@@ -130,3 +130,25 @@ resource "aws_cloudwatch_metric_alarm" "database_memory_free" {
+
+resource "aws_cloudwatch_metric_alarm" "database_cpu_credits" {
+  // This results in 1 if instance_type starts with "db.t", 0 otherwise.
+  count               = "${replace(replace(var.instance_type, "/^db\\.[^t].*/", "0"), "/^db\\.t.*$/", "1")}"
```
I have yet to use this interpolation method, but have you tried what's described here? https://www.hashicorp.com/blog/terraform-0-8/#conditional

```hcl
variable "env" { default = "development" }

resource "aws_instance" "foo" {
  count = "${var.env == "production" ? 1 : 0}"
}
```

Perhaps we can use `instance_type` in the conditional to go down one branch vs. the other with `contains()` or `split()`? Mostly just trying to make it easier to digest for someone who tries to read this a few weeks or month down the road.

I'm also cool with making this module Terraform 0.8+ only.

dagvl commented:
@hectcastro I've updated it using terraform 0.8 conditionals now. I was unaware of those, and it ended up a lot more readable. Thanks!  

I also added a version requirement to the module of >= 0.8.0 due to the conditional.

hectcastro commented on main.tf:
```diff
@@ -1,4 +1,11 @@
+# Terraform required version
+#
+terraform {
+  required_version = ">= 0.8.0"
```
Nice. I was actually unaware of this.
